Lee/fix login

Hyukjoo-Lee opened the pull request:
## Description

- I have resolved a login issue where the page would continue loading even with invalid user information. 

## Solution
- By utilizing the 'response.ok' value, I have checked if user information is correct or not. 
- If the information is incorrect, a toast message is displayed, and the form is not submitted by setting 'setSubmitForm: false'

## Screen
<img width="500" alt="Screen Shot 2023-08-05 at 3 04 15 AM" src="https://github.com/HanKyungSung/inviteyou/assets/96518885/efe235c7-f3c4-4e9d-b4fc-2e1fea06e9c9">

Hyukjoo-Lee committed:
fixed login function

Hyukjoo-Lee committed:
install react-toastify library

Hyukjoo-Lee committed:
just removed a comment and annotations.

HanKyungSung commented on invitation-frontend/package.json:
```diff
@@ -1,75 +1,76 @@
+    "react-responsive-carousel": "^3.2.23",
+    "react-router-dom": "^6.4.2",
+    "react-swiper": "^0.1.5",
+    "react-toastify": "^9.1.3",
```
Is there a specific reason why you installed a new package?

Have you tried to look at this? [https://mantine.dev/core/notification/](https://mantine.dev/core/notification/)?

Hyukjoo-Lee committed:
intermediate commit

Hyukjoo-Lee committed:
reserve conflict merge from main to lee/fix-login

Hyukjoo-Lee committed:
fixed login function

Hyukjoo-Lee commented:
I have employed Mantine in place of a new library. 
Could you please review and confirm if we can proceed with merging?

HanKyungSung commented:
![image](https://github.com/HanKyungSung/inviteyou/assets/25420417/1682fc46-0e21-4001-9d63-2adeb8149bfd)

As the screenshot above, there are a few things we need to adjust.

- The notification box won't close even clicking the x button.
- The email and password box border should turn red to indicate either box is wrong.
  > Make sure to use the Mantine component props. I believe it is an `error` prop but double-check before using it.

Hyukjoo-Lee committed:
fixed notification's close button function

Hyukjoo-Lee committed:
Update error message handling

Hyukjoo-Lee committed:
Created indication for which input is incorrect

Hyukjoo-Lee commented:
### Description
- I have fixed an issue with the close button for the Notification component.
- I have implemented error indication for incorrect login inputs.

### Solution
- To close a Notification I added 'setNotificationVisible(true)' 
- Modified specific HTTP response status codes to identify incorrect inputs. 
response.status === 404, indicated an incorrect email
response.status === 403, indicated an incorrect password
- To indicate which input field is incorrect, I modified the handleShowNotification function, passing values (error messages and fields (either 'email' or 'password')

### Screen
<img width="491" alt="Screen Shot 2023-08-15 at 2 25 59 AM" src="https://github.com/HanKyungSung/inviteyou/assets/96518885/1a94ac50-7969-41e1-b097-418e81b2e5fd">

<img width="479" alt="Screen Shot 2023-08-15 at 2 27 09 AM" src="https://github.com/HanKyungSung/inviteyou/assets/96518885/d380218b-bf26-41e2-a5f6-caca9975a9bb">

Hyukjoo-Lee committed:
removed nested if-else statement
